spring-integration-core/src/main/java/org/springframework/integration/scattergather/ScatterGatherHandler.java

```diff
@@ -167,9 +165,7 @@ else if (this.gatherChannel instanceof ReactiveStreamsSubscribableChannel) {
 		MessageHeaders headers = message.getHeaders();
 		return getMessageBuilderFactory()
 				.fromMessage(message)
-				.setHeader(MessageHeaders.REPLY_CHANNEL, headers.get(ORIGINAL_REPLY_CHANNEL))
-				.setHeader(MessageHeaders.ERROR_CHANNEL, headers.get(ORIGINAL_ERROR_CHANNEL))
-				.removeHeaders(ORIGINAL_REPLY_CHANNEL, ORIGINAL_ERROR_CHANNEL)
+				.setHeader(MessageHeaders.ERROR_CHANNEL, headers.get(GATHER_RESULT_CHANNEL))
```

Comment:
I find this as a wrong decision. We have that `errorChannelName` to process exceptions in the scatterer sub-flows and gatherer.

Reply:
@artembilan,
errorChannelName  is meant to handle errors in the subflows. The result will flow back to Gather Channel.
If there is an error the Gatherer and if we propagate the error back to errorChannelName, it will cause the Scatter Gather to hang for ever if any exception thrown in outputProcessor.